core/src/main/kotlin/org/evomaster/core/output/naming/RestActionTestCaseNamingStrategy.kt

```diff
@@ -123,17 +147,14 @@ open class RestActionTestCaseNamingStrategy(
      * The filter call ensures that we are only performing this disambiguation when there's only one individual that
      * differs and the list of query params is not empty.
      */
-    private fun solveQueryParamsAmbiguities(duplicatedIndividuals: MutableSet<EvaluatedIndividual<*>>): Map<EvaluatedIndividual<*>, String> {
+    private fun getQueryParamsDisambiguationIndividuals(duplicatedIndividuals: MutableSet<EvaluatedIndividual<*>>): List<EvaluatedIndividual<*>> {
         return duplicatedIndividuals
             .groupBy {
-                (it.evaluatedMainActions().last().action as RestCallAction).parameters.filterIsInstance<QueryParam>()
+                val restAction = it.evaluatedMainActions().last().action as RestCallAction
+                restAction.path.getOnlyQuery(restAction.parameters)
             }
-            .filter { it.value.size == 1 && it.key.isNotEmpty()}
-            .mapNotNull { entry ->
-                val eInd = entry.value[0]
-                eInd to expandName(eInd, mutableListOf(), ::queryParamsAmbiguitySolver)
-            }
-            .toMap()
+            .filter { it.value.isNotEmpty() && it.key.isNotEmpty()}
+            .flatMap { it.value }.toList()
```

Comment:
best to have each stream operation on its own line, ie, in this case the `.toList()` can go on new line